### PR TITLE
#809 implement pandas 2.0 upgrade

### PR DESCRIPTION
**Pull request by SarahJohnsonONS**
Pandas updated to version 2.0.2 and failing unit/behaviour tests fixed. New test case created for multi unit single measure data set as existing one was single unit single measure. GH Action now incorporates testing on Python 3.11/Ubuntu for pandas ^1.3 and latest version.

### CODE REVIEW

**Review by nimshi89 — APPROVED** [review]
Looks fine

**Review by canwaf — COMMENTED** [review]
@SarahJohnsonONS, can you walk me through the reasons why we're updating and introducing new test cases?

**Review by canwaf — APPROVED** [review]

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -7,15 +7,18 @@ jobs:
     strategy:
       matrix:
         # Test on linux with the oldest and newest versions of python.
```
Comment (robons) [inline .github/workflows/pull-request.yaml:9]: This comment no longer reflects the truth.

File: .github/workflows/reusable-test.yaml [hunk 2]
```diff
@@ -61,6 +73,10 @@ jobs:
       - name: Install python packages
         run: poetry install --sync
 
+      - name: Specify pandas version (Ubuntu & Python 3.11)
+        if: inputs.os == 'ubuntu-latest' && inputs.python-version == '3.11'
```
Comment (robons) [inline .github/workflows/reusable-test.yaml:77]: Hmmmmmm I think this should be done under all conditions.

File: .github/workflows/reusable-test.yaml [hunk 3]
```diff
@@ -52,6 +55,15 @@ jobs:
...
       - uses: actions/cache@v3
+        if: inputs.os == 'ubuntu-latest' && inputs.python-version == '3.11'
+        with:
+          path: |
+            ~/.cache/pypoetry/virtualenvs
+            C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs
+          key: ${{ inputs.os }}-${{ steps.install-python.outputs.python-version }}-poetry-${{ hashFiles('poetry.lock') }}-${{ inputs.pandas-version }}
+
+      - uses: actions/cache@v3
+        if: (inputs.os == 'ubuntu-latest' && inputs.python-version == '3.9') || inputs.os == 'windows-latest'
```
Comment (robons) [inline .github/workflows/reusable-test.yaml:66]: I'm not really happy about these if statements here. It'd be better if they weren't aware of which versions of things the external caller is using. 

Why is this necessary here? We should really have a comment  explaining that.

File: tests/behaviour/inspectcli.feature [hunk 4]
```diff
@@ -143,175 +143,94 @@ Feature: Behaviour testing of csvcubed inspect.
...
+          - Contact Point: None
+          - Identifier: Multi unit single measure dataset
+          - Comment: None
+          - Description: None
         """
         And the Data Structure Definition Printable should be
         """
         - The data cube has the following column component information:
-            - Dataset Label: Final UK greenhouse gas emissions national statistics: 1990 to 2019
+            - Dataset Label: Multi unit single measure dataset
```
Comment (robons) [inline tests/behaviour/inspectcli.feature:178]: Why did this input need to change?
